src/Castle.Windsor.Extensions.DependencyInjection/Extensions/ServiceDescriptorExtensions.cs

```diff
@@ -20,6 +20,8 @@ namespace Castle.Windsor.Extensions.DependencyInjection.Extensions
 
 	public static class ServiceDescriptorExtensions
 	{
+		private static MethodInfo fromServiceDescriptorMethod = typeof(RegistrationAdapter).GetMethod(nameof(RegistrationAdapter.FromServiceDescriptor), BindingFlags.Static | BindingFlags.Public);
```

Comment:
If we do go ahead with this rather than #555, then this should be marked readonly and probably follow the pattern we use throughout Castle.

e.g.:

```csharp
/// <summary>
/// Holds <see cref="MethodInfo" /> objects representing methods of the <see cref="RegistrationAdapter" /> class.
/// </summary>
internal static class RegistrationAdapterMethods
{
	public static readonly MethodInfo FromServiceDescriptor =
		typeof(RegistrationAdapter).GetMethod(nameof(RegistrationAdapter.FromServiceDescriptor), BindingFlags.Static | BindingFlags.Public);
}
```

Reply:
Done.